Flashcards/task/src/flashcards/Main.java

```diff
@@ -47,133 +90,248 @@ private void runGameByAction() {
+                } else {
+                    flashcard.setDefinition(cardDefinition);
+                    flashcard.setErrors(Integer.parseInt(cardErrors));
+                }
                 cardsImported++;
             }
-            System.out.println(cardsImported + " cards have been loaded.");
+            outputMsgAndLog(cardsImported + " cards have been loaded.");
         } catch (IOException e) {
-            System.out.println("File not found.");
+            outputMsgAndLog("File not found.");
         }
     }
 
     private void exportCards() {
-        System.out.println("File name:");
+        outputMsgAndLog("File name:");
         File file = new File(userScanner.nextLine());
+        applicationLogger(file.toString());
 
         int numberOfSavedCards = 0;
 
         try (FileWriter cardWriter = new FileWriter(file, false)) {
-            for (String cardQuestion : cardMap.keySet()) {
-                cardWriter.write(cardQuestion + "\n");
-                cardWriter.write(cardMap.get(cardQuestion) + "\n");
+            for (Flashcard flashcard : flashcardList) {
+                cardWriter.write(flashcard.getQuestion() + "\n");
+                cardWriter.write(flashcard.getDefinition() + "\n");
+                cardWriter.write(flashcard.getErrors() + "\n");
                 numberOfSavedCards++;
             }
         } catch (IOException e) {
-            System.out.println("Something went wrong: \n" + e);
-        }
-        System.out.println(numberOfSavedCards + " cards have been saved.");
-    }
-
-    private void fillCardMap() {
-        System.out.println("The card:");
-        String cardQuestion = userScanner.nextLine();
-
-        if (cardMap.containsKey(cardQuestion)) {
-            System.out.println("The card \"" + cardQuestion + "\" already exists.");
-            return;
-        }
-
-        System.out.println("The definition of the card:");
-        String cardDefinition = userScanner.nextLine();
-
-        if (cardMap.containsValue(cardDefinition)) {
-            System.out.println("The definition \"" + cardDefinition + "\" already exists.");
-            return;
-        }
-
-        cardMap.put(cardQuestion, cardDefinition);
-        System.out.println("The pair (\"" + cardQuestion + "\"" + ":" + "\"" + cardDefinition + "\") has been added.");
-    }
-
-    private String getRightCardQuestion(String answer) {
-        for (String cardQuestionToFind : cardMap.keySet()) {
-            if (answer.equals(cardMap.get(cardQuestionToFind))) {
-                return cardQuestionToFind;
-            }
+            outputMsgAndLog("Something went wrong: \n" + e);
         }
-        return null;
+        outputMsgAndLog(numberOfSavedCards + " cards have been saved.");
     }
 
-    private String checkAnswer(String cardQuestion, String cardAnswer) {
-        if (cardMap.get(cardQuestion).equals(cardAnswer)) {
+    private String checkAnswer(Flashcard playedFlashcard, String cardAnswer) {
```

Comment:
Deze methode kan verbetert worden! Probeer het controleren van het antwoord in je object te verwerken. En let op dat je geen regels code dubbel schrijft

Reply:
Het verwerken in het object is niet gebeurd, ik heb het wel aangepast. 
Ik snap nog niet helemaal waarom het in het object moet als de list met cards buiten het object leeft. Dan kan de methode om die lijst door te zoeken er toch ook buiten? Wellicht iets om over te bomen nog.